web/modules/custom/joinup_group/src/Controller/GroupMembershipPermissionsInformationController.php

```diff
@@ -0,0 +1,259 @@
+          'facilitator' => FALSE,
+          'owner' => FALSE,
+        ],
+      ],
+      [
+        'description' => $this->t('Approve requested deletion of content'),
+        'applicable' => $is_moderated && !$only_authors_can_create,
+        'permitted' => [
+          'member' => FALSE,
+          'author' => FALSE,
+          'facilitator' => TRUE,
+          'owner' => TRUE,
+        ],
+      ],
+      [
+        'description' => $is_moderated && !$only_authors_can_create ? $this->t('Delete own content without approval') : $this->t('Delete own content'),
+        'permitted' => [
+          'member' => !$is_moderated && !$only_authors_can_create,
+          'author' => TRUE,
+          'facilitator' => TRUE,
+          'owner' => TRUE,
+        ],
+      ],
+      [
+        'description' => $this->t('Delete any content'),
+        'permitted' => [
+          'member' => FALSE,
+          'author' => FALSE,
+          'facilitator' => TRUE,
+          'owner' => TRUE,
+        ],
+      ],
+      [
+        'description' => $this->t('Manage users'),
+        'permitted' => [
+          'member' => FALSE,
+          'author' => FALSE,
+          'facilitator' => FALSE,
+          'owner' => TRUE,
+        ],
+      ],
+    ];
+
+    $build['permissions'] = [
+      '#type' => 'table',
+      '#header' => [
+        $this->t('Permission'),
+        $this->t('Member'),
+        $this->t('Author'),
+        $this->t('Facilitator'),
+        $this->t('Owner'),
+      ],
+      '#attributes' => ['class' => ['form-table__member-permissions']],
+    ];
+
+    foreach ($permissions_info as $permission_info) {
+      if ($permission_info['applicable'] ?? TRUE) {
+        $build['permissions'][] = [
+          [
+            '#plain_text' => $permission_info['description'],
+            '#wrapper_attributes' => [
+              'class' => ['form-table__cell__description'],
+            ],
+          ],
+          [
+            '#plain_text' => $permission_info['permitted']['member'] ? '✓' : '',
+            '#wrapper_attributes' => ['class' => ['form-table__cell__data']],
+          ],
+          [
+            '#plain_text' => $permission_info['permitted']['author'] ? '✓' : '',
+            '#wrapper_attributes' => ['class' => ['form-table__cell__data']],
+          ],
+          [
+            '#plain_text' => $permission_info['permitted']['facilitator'] ? '✓' : '',
+            '#wrapper_attributes' => ['class' => ['form-table__cell__data']],
+          ],
+          [
+            '#plain_text' => $permission_info['permitted']['owner'] ? '✓' : '',
+            '#wrapper_attributes' => ['class' => ['form-table__cell__data']],
+          ],
+        ];
+      }
+    }
+
+    $build['close'] = [
+      '#type' => 'link',
+      '#title' => $this->t('Got it'),
+      '#url' => Url::fromRoute('entity.rdf_entity.member_overview', [
+        'rdf_entity' => $rdf_entity->id(),
+      ]),
+      '#attributes' => [
+        'class' => [
```

Comment:
According to what we have said in the past, all these should go to the theme, not hardcoded in the controller. The same applies for all classes above with the table cell data.